aten/src/TH/THRandom.cpp

```diff
@@ -13,11 +13,12 @@
 /* Creates (unseeded) new generator*/
 static THGenerator* THGenerator_newUnseeded()
 {
-  THGenerator *self = THAlloc(sizeof(THGenerator));
+  THGenerator *self = (THGenerator *)THAlloc(sizeof(THGenerator));
   memset(self, 0, sizeof(THGenerator));
   self->left = 1;
   self->seeded = 0;
   self->normal_is_valid = 0;
+  self->mutex = new std::mutex();
```

Comment:
It would be nicer to use placement new (no extra indirection):
```cpp
new (&self->mutex) std::mutex();
```
This requires you to put this in the destructor:
```cpp
self->mutex.~mutex();
```